challenge.rb

```diff
@@ -1,26 +1,34 @@
 
 def capitalize_each_string(input)
   #implement your solution here
+  input.map do |i|
+    i.capitalize
+  end
 end
 
 def fetch_the_dog(input)
   #implement your solution here
+  input.select {|i| i == "dog"}
 end
 
 def no_dogs_allowed(input)
   #implement your solution here
+  input.reject {|i| i == "dog"}
 end
 
 def count_the_animals(input)
   #implement your solution here
+    input.length
 end
 
 def fetch_the_first_two(input)
   #implement your solution here
+  input.first(2)
 end
 
 def fetch_CD_animals(input)
   #implement your solution here
+  input.find_all {|i| i.start_with?("c") || i.start_with?("d")}
```

Comment:
cool. you can also use `.grep` which takes a regex pattern. it's super handy.
https://ruby-doc.org/core-2.5.0/Enumerable.html#method-i-grep